src/rdkafka_telemetry_decode.c

```diff
@@ -0,0 +1,543 @@
+
+        pb_istream_t stream              = pb_istream_from_buffer(buffer, size);
+        metricsData.resource_metrics.arg = decode_interface;
+        metricsData.resource_metrics.funcs.decode = &decode_resource_metrics;
+
+        bool status = pb_decode(
+            &stream, opentelemetry_proto_metrics_v1_MetricsData_fields,
+            &metricsData);
+        if (!status) {
+                RD_INTERFACE_CALL(decode_interface, decode_error,
+                                  "Failed to decode MetricsData: %s",
+                                  PB_GET_ERROR(&stream));
+        }
+        return status;
+}
+
+static void unit_test_telemetry_decoded_string(void *opaque,
+                                               const uint8_t *decoded) {
+        switch (unit_test_data.current_field) {
+        case 2:
+                rd_snprintf(unit_test_data.metric_name,
+                            sizeof(unit_test_data.metric_name), "%s", decoded);
+                break;
+        case 3:
+                rd_snprintf(unit_test_data.metric_description,
+                            sizeof(unit_test_data.metric_description), "%s",
+                            decoded);
+                break;
+        default:
+                break;
+        }
+        unit_test_data.current_field++;
+}
+
+static void unit_test_telemetry_decoded_NumberDataPoint(
+    void *opaque,
+    const opentelemetry_proto_metrics_v1_NumberDataPoint *decoded) {
+        unit_test_data.metric_value_int    = decoded->value.as_int;
+        unit_test_data.metric_value_double = decoded->value.as_double;
+        unit_test_data.metric_time         = decoded->time_unix_nano;
+        unit_test_data.current_field++;
+}
+
+static void
+unit_test_telemetry_decoded_type(void *opaque,
+                                 rd_kafka_telemetry_metric_type_t type) {
+        unit_test_data.type = type;
+        unit_test_data.current_field++;
+}
+
+static void unit_test_telemetry_error(void *opaque, const char *error, ...) {
+        char buffer[1024];
+        va_list ap;
+        va_start(ap, error);
+        rd_vsnprintf(buffer, sizeof(buffer), error, ap);
+        va_end(ap);
+        RD_UT_SAY("%s", buffer);
+}
+
+bool unit_test_telemetry(rd_kafka_telemetry_producer_metric_name_t metric_name,
+                         const char *expected_name,
+                         const char *expected_description,
+                         rd_kafka_telemetry_metric_type_t expected_type,
+                         rd_bool_t is_double) {
+        rd_kafka_t *rk = rd_calloc(1, sizeof(*rk));
+        rwlock_init(&rk->rk_lock);
+        rk->rk_type                          = RD_KAFKA_PRODUCER;
+        rk->rk_telemetry.matched_metrics_cnt = 1;
+        rk->rk_telemetry.matched_metrics =
+            rd_malloc(sizeof(rd_kafka_telemetry_producer_metric_name_t) *
+                      rk->rk_telemetry.matched_metrics_cnt);
+        rk->rk_telemetry.matched_metrics[0] = metric_name;
```

Comment:
Add these so the `producer.connection.creation.rate` becomes 1.0 too
```suggestion
        rk->rk_telemetry.matched_metrics[0] = metric_name;
        rk->rk_telemetry.rk_historic_c.ts_start =
            (rd_uclock() - 1000 * 1000) * 1000;
        rk->rk_telemetry.rk_historic_c.ts_last =
            (rd_uclock() - 1000 * 1000) * 1000;
```

Reply:
Updated